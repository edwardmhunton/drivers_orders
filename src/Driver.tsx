import React from "react";

import { Typography, Card } from "@material-ui/core";

const styles = {
  width: 300,
  height: 50,
  fontSize: 20,
  display: "flex",
  flexDirection: "column",
  margin: 10,
  textAlign: "left"
};

const liveDriver = {
  color: "#6991FD"
};
const nonLiveDriver = {
  color: "#FD7467"
};

export const Driver = driver => {
  const mergedStyles = driver.proximity
    ? Object.assign({}, styles, liveDriver)
    : Object.assign({}, styles, nonLiveDriver);
  return (
    <Card className={"badge badge-secondary"}>
      <Typography style={mergedStyles}>{`Driver ID: ${driver.id}`}</Typography>
    </Card>
  );
};
