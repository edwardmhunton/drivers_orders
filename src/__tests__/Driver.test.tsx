import React from "react";

import renderer from "react-test-renderer";

import { Driver } from "../Driver";

describe("the fist test", () => {
  const driver = {
    id: "8376587365",
    coordinates: { lat: "51.57131", lng: "-0.1048" },
    proximity: false,
    distance: 6500
  };
  it("Renders Correctly", () => {
    const DriverComponent = renderer
      .create(<Driver driver={driver} />)
      .toJSON();
    expect(DriverComponent).toMatchSnapshot();
  });
});
