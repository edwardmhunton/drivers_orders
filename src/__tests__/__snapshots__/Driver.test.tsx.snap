// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver test Renders Correctly 1`] = `
<div
  className="badge badge-secondary"
  style={
    Object {
      "color": "red",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": 20,
      "height": 50,
      "margin": 10,
      "textAlign": "left",
      "width": 300,
    }
  }
>
  Driver ID: undefined
</div>
`;
